save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt

```diff
@@ -11,8 +11,12 @@ import okio.Path
 /**
  * Plugin that can be injected into SAVE during execution. Plugins accept contents of configuration file and then perform some work.
  * @property testConfig
+ * @property useInternalRedirections whether to redirect stdout/stderr for internal purposes
  */
-abstract class Plugin(open val testConfig: TestConfig, private val testFiles: List<String>) {
+abstract class Plugin(
+    open val testConfig: TestConfig,
+    private val testFiles: List<String>,
+    protected val useInternalRedirections: Boolean) {
```

Comment:
In which case would we need to change this parameter? Does it make sense to make it configurable from plugins, when extending this class, or can it be configured globally application-wise?

Reply:
We need to run tests with `useInternalRedirections = false`, since we use `echo >` in them.
In other cases we use `useInternalRedirections = true`, for now. 

Previously it was hard coded for FixPlugin to `false`, which is incorrect, since we need to collect `stderr` and `stdout` in the course of work, so I decide to make it configurable.

In general, this argument just a workaround, since we use redirections for two purposes:
1) collect process output, by creating temp files
2) creating new files with some info (like we do in tests)
and unfortunately we can't do this at the same time. Only the last redirection in bash command will be applied.

So, the configuration must be hidden or carried out with full awareness of the consequences

Reply:
I'd prefer to hide it from plugin implementations, however it seems tricky to do. We could make it `private` in `Plugin` and also create `useInternalRedirections` in `ProcessBuilder` itself (because we won't need to change it between invocations of the same ProcessBuilder, if I understand you correctly). Then we can move `val pb = ProcessBuilder(useInternalRedirections)` to the base `Plugin` class, so the only place, where plugin implementation sees this parameter is constructor invocation. What do you think?

Reply:
Yes, thanks, it will be even better. Changed